docs/tutorials/tutorial_2.md

```diff
@@ -0,0 +1,177 @@
+# Tutorial 2 - Implement custom components
+
+**Level**: Beginner
+**Time**: 20min
+**Goal**: After completing this tutorial with Fondant, you will be able to build your own custom component and integrate it into a fondant pipeline. 
+
+**Prerequisite**: Make sure docker compose is installed on your local system.
+We recommend completing the first tutorial before proceeding with this one, as this tutorial builds upon the knowledge gained in the previous one.
+
+## Overview
+
+In the initial tutorial, you learned how to create your first Fondant pipeline. While the example demonstrates initialising the dataset from HuggingFace and using a reusable component to download images, this is just the beginning.
+
+The true power of Fondant lies in its capability to enable you to construct your own data pipelines to create high-quality datasets. To achieve this, we need to implement custom components.
+
+In this tutorial, we will guide you through the process of implementing your very own custom component. We will illustrate this by demonstrating how to filter images based on file type.
+
+This pipeline is an extension of the one introduced in the first tutorial. After loading the dataset from HuggingFace, it narrows down the dataset to exclusively encompass PNG files. Finally, it proceeds to download these carefully selected images.
+
+
+## Build your custom component
```

Comment:
Shouldn't the 2 following sections (Creating the ComponentSpec, Creating the Component, Build the component) be subsections of this section?